packages/kit/src/exports/vite/graph_analysis/index.js

```diff
@@ -18,6 +19,10 @@ const ILLEGAL_MODULE_NAME_PATTERN = /.*\.server\..+/;
 export function is_illegal(id, dirs) {
 	if (ILLEGAL_IMPORTS.has(id)) return true;
 	if (!id.startsWith(dirs.cwd) || id.startsWith(dirs.node_modules)) return false;
+	
+	// Allow remote functions to be imported on the client side (they get transformed)
+	if (REMOTE_FILE_PATTERN.test(path.basename(id))) return false;
+	
```

Comment:
I didn't find how `lib` and `routes` are doing this!

_I'm probably missing something!_